src/lib/serialization/schema.ts

```diff
@@ -303,19 +293,17 @@ export interface MappedType
 
 export interface TypeOperatorType
     extends Type,
-        S<M.TypeOperatorType, "operator" | "target"> {}
+        S<M.TypeOperatorType, "type" | "operator" | "target"> {}
 
-export interface UnionType extends Type, S<M.UnionType, "types"> {}
+export interface UnionType extends Type, S<M.UnionType, "type" | "types"> {}
 
-export interface UnknownType extends Type, S<M.UnknownType, "name"> {}
+export interface UnknownType extends Type, S<M.UnknownType, "type" | "name"> {}
 
 /**
  * Technically not correct, the `type` property will be set by the abstract serializer.
  * But to allow tagged literals, the `type` property is instead defined by each child type.
  */
-export interface Type {
-    type: keyof TypeKindMap;
-}
+export interface Type {}
```

Comment:
The comment can go away now - I'd like to keep the interface though so that it's available as a target for declaration merging (don't think anyone will use it, but....)

Reply:
Removed the comment. I am a bit split about keeping `Type` — given that TS has a structural type system, keeping an empty interface feels redundant. But let's leave it for now.